.github/workflows/kubeflex-compatibility.yml

```diff
@@ -0,0 +1,126 @@
+name: KubeFlex Compatibility Matrix
+
+on:
+  schedule:
+    - cron: '0 4 * * *'
+  workflow_dispatch:
+    inputs:
+      kubestellar_ref:
+        description: 'KubeStellar git ref (branch, tag, or commit)'
+        required: true
+        default: 'main'
+
+permissions:
+  contents: read
+
+jobs:
+  setup-and-deploy:
+    name: Setup Environment and Deploy KubeStellar
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          ref: ${{ github.event.inputs.kubestellar_ref || 'main' }}
+
+      - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5
+        with:
+          go-version: '1.24'
+          cache: true
+
+      - name: Install kubectl
+        uses: azure/setup-kubectl@776406bce94f63e41d621b960d78ee25c8b76ede
+
+      - uses: ko-build/setup-ko@d006021bd0c28d1ce33a07e7943d48b079944c8d
+
+      - name: Install common dependencies
+        run: |
+          bash <(curl -L https://raw.githubusercontent.com/open-cluster-management-io/clusteradm/main/install.sh)
+          go install github.com/onsi/ginkgo/v2/ginkgo
+
+      - name: Setup KubeStellar
+        run: |
+          cd test/e2e/
+          ./common/setup-kubestellar.sh
+
+      - name: Upload kubeconfig
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02
+        with:
+          name: kubeconfig
+          path: ~/.kube/config
+
+  generate-matrix:
+    name: Generate KubeFlex version matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.generate.outputs.matrix }}
+    steps:
+      - name: Generate kflex version matrix
+        id: generate
+        run: |
+          MIN_VERSION="v0.8.0"
+          MATRIX_VERSIONS=$(curl -s "https://api.github.com/repos/kubestellar/kubeflex/releases" | jq -r --arg MIN_VERSION "$MIN_VERSION" '[.[] | .tag_name | select(test("^v[0-9]+\\.[0-9]+\\.[0-9]+$")) | select(. >= $MIN_VERSION)]')
+          echo "matrix=$MATRIX_VERSIONS" >> $GITHUB_OUTPUT
+
+  test-kubeflex-version:
+    name: Test with kflex ${{ matrix.kflex_version }}
+    needs: [setup-and-deploy, generate-matrix]
+    runs-on: ubuntu-latest
```

Comment:
Shall I infer that this job is run on a fresh worker, the state from the previous jobs being gone with their completion?

Reply:
https://docs.github.com/en/actions/get-started/understand-github-actions#the-components-of-github-actions says

> Each job will run inside its own virtual machine runner, or inside a container